src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -1138,13 +1814,16 @@ subroutine refract(En, cn, freq, dt, G, US, NAngle, use_PPMang)
                  (G%CoriolisBu(I,J-1)**2 + G%CoriolisBu(I-1,J)**2))
       favg = 0.25*((G%CoriolisBu(I,J) + G%CoriolisBu(I-1,J-1)) + &
                    (G%CoriolisBu(I,J-1) + G%CoriolisBu(I-1,J)))
-      df_dx = 0.5*((G%CoriolisBu(I,J) + G%CoriolisBu(I,J-1)) - &
-                    (G%CoriolisBu(I-1,J) + G%CoriolisBu(I-1,J-1))) * G%IdxT(i,j)
-      df_dy = 0.5*((G%CoriolisBu(I,J) + G%CoriolisBu(I-1,J)) - &
-                   (G%CoriolisBu(I,J-1) + G%CoriolisBu(I-1,J-1))) * G%IdyT(i,j)
+      df_dx = 0.5*G%IdxT(i,j)*((G%CoriolisBu(I,J) - G%CoriolisBu(I-1,J-1)) + &
+                               (G%CoriolisBu(I,J-1) - G%CoriolisBu(I-1,J)))
+      df_dy = 0.5*G%IdyT(i,j)*((G%CoriolisBu(I,J) - G%CoriolisBu(I-1,J-1)) + &
+                               (G%CoriolisBu(I-1,J) - G%CoriolisBu(I,J-1)))
 
-      dlnCn_dx = G%IdxT(i,j) * (cn_u(I,j) - cn_u(I-1,j)) / (0.5 * (cn_u(I,j) + cn_u(I-1,j)) + cn_subRO)
-      dlnCn_dy = G%IdyT(i,j) * (cn_v(i,J) - cn_v(i,J-1)) / (0.5 * (cn_v(i,J) + cn_v(i,J-1)) + cn_subRO)
+      !dlnCn_dx = G%IdxT(i,j) * (cn_u(I,j) - cn_u(I-1,j)) / (0.5 * (cn_u(I,j) + cn_u(I-1,j)) + cn_subRO)
+      !dlnCn_dy = G%IdyT(i,j) * (cn_v(i,J) - cn_v(i,J-1)) / (0.5 * (cn_v(i,J) + cn_v(i,J-1)) + cn_subRO)
+      ! to avoid substractions in cn, this can be rewritten as:
+      dlnCn_dx = 2*G%IdxT(i,j)*( (2*cn_u(I,j))/(cn_u(I,j) + cn_u(I-1,j) + cn_subRO) - 1.0 )
```

Comment:
Although this expression is mathematically correct and (almost) equivalent to the expression two lines above, it is not symmetric in the x-direction, and hence will (slightly) break rotational symmetry.  Moreover, even though it is not subtracting adjacent values of cn_u, this does not help with roundoff because there is the same magnitude of truncation errors arising from subtraction of the fraction in line 1825 from 1.   I would suggest replacing lines 1825 and 1826 by the expressions on lines 1822 and 1823.

Reply:
done